### PR TITLE
feat(knative): bump to 1.15.2

### PR DESCRIPTION
**Pull request by mhrabovcin**
**What type of PR is this?**

Feat

**What this PR does/ why we need it**:

Creates new knative chart with latest 1.15.2 version.

**Which issue(s) this PR fixes**:

https://jira.nutanix.com/browse/NCN-102804

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Review by deepakm-ntnx — APPROVED** [review]
could you please document the upstream location from where the template files are copied. would be good to add a script which one can run to download so that future upgrades are automated as much as possible.

File: staging/knative/upgrade_knative.sh [hunk 1]
```diff
@@ -21,6 +21,9 @@ PATCH_1_FIX=$'# eg. \'{{ `{{.Name}}-{{.Namespace}}.{{ index .Annotations "sub"}}
...
 PATCH_2_FIX=$'logging.request-log-template: \'{"httpRequest": {"requestMethod": "{{ `{{.Request.Method}}", "requestUrl": "{{js .Request.RequestURI}}", "requestSize": "{{.Request.ContentLength}}", "status": {{.Response.Code}}, "responseSize": "{{.Response.Size}}", "userAgent": "{{js .Request.UserAgent}}", "remoteIp": "{{js .Request.RemoteAddr}}", "serverIp": "{{.Revision.PodIP}}", "referer": "{{js .Request.Referer}}", "latency": "{{.Response.Latency}}s", "protocol": "{{.Request.Proto}}"}, "traceId": "{{index .Request.Header "X-B3-Traceid"}}` }}"}\''
 
+PATCH_3=$'#               serving.knative.dev/revision: {{revision-name}}'
+PATCH_3_FIX=$'#               serving.knative.dev/revision: {{ `{{revision-name}}` }}'
+
 # Base URLs
 SERVING_URL=https://github.com/knative/serving/releases/download/knative-v${SERVING_TAG}
```
Comment (deepakm-ntnx) [inline staging/knative/upgrade_knative.sh:28]: nvm, this is the upgrade location. found it

**Comment by deepakm-ntnx** [comment]
> could you please document the upstream location from where the template files are copied. would be good to add a script which one can run to download so that future upgrades are automated as much as possible.

found it. https://github.com/mesosphere/charts/pull/1517/files#r1795775689
can we document the step by step process one does in order to get this upgrade done.
1. update versions in upgrade_knative.sh
2. run upgrade_knative.sh
3. commit?

**Comment by mhrabovcin** [comment]
@deepakm-ntnx we already have document for that https://github.com/mesosphere/charts/blob/master/staging/knative/UPGRADING.md .

**Comment by mhrabovcin** [comment]
> can we point or list all the tests we run on such upgrade.

@deepakm-ntnx I was testing the upgrade manually. I've found 2 issues:
1. validating webhook was failing becuase of the presence of default `_example` keys in `ConfigMap` resources. I've added a patch that will remove these keys when building YAML documents.
2. the `helm upgrade -i ...` was failing because of TLS verification issues between k8s API server and webhook container. I was able to run the upgrade by removing the validating webhook. I think we can add it as a release note. I don't think we have any customers running `knative` right now.

**Review by gracedo — APPROVED** [review]

**Pull request merged**